src/Microsoft.AspNetCore.Hosting/WebHostExtensions.cs

```diff
@@ -90,8 +89,7 @@ public static void Run(this IWebHost host)
                     done.Wait();
                 };
 
-                var assemblyLoadContext = AssemblyLoadContext.GetLoadContext(typeof(WebHostExtensions).GetTypeInfo().Assembly);
-                assemblyLoadContext.Unloading += context => shutdown();
+                AppDomain.CurrentDomain.ProcessExit += (sender, eventArgs) => shutdown();
```

Comment:
It is being tested here: https://github.com/aspnet/Hosting/blob/dev/test/Microsoft.AspNetCore.Hosting.FunctionalTests/ShutdownTests.cs#L25. I asked @pakrym to do some more in depth manual testing to verify the behaviour later though. I would prefer getting this change in with this PR so we don't cross-compile.